ext/standard/assert.c

```diff
@@ -139,17 +139,20 @@ PHP_MINFO_FUNCTION(assert) /* {{{ */
    Checks if assertion is false */
 PHP_FUNCTION(assert)
 {
-	zval *assertion;
-	zval *description = NULL;
+	/* `desc_or_throwable` may either be a description string which will be used in
+	 * the message for a failed assertion, or a `Throwable` object which will be
+	 * thrown if the assertion fails */
+	zval *assertion, *desc_or_throwable = NULL, *assertion_str = NULL;
 
 	if (! ASSERTG(active)) {
 		RETURN_TRUE;
 	}
 
-	ZEND_PARSE_PARAMETERS_START(1, 2)
+	ZEND_PARSE_PARAMETERS_START(1, 3)
 		Z_PARAM_ZVAL(assertion)
 		Z_PARAM_OPTIONAL
-		Z_PARAM_ZVAL(description)
+		Z_PARAM_ZVAL(desc_or_throwable)
+		Z_PARAM_ZVAL(assertion_str) /* inserted into call by `zend_compile_assert` */
```

Comment:
This complicates API. I don't think, this makes sense.

Reply:
Indeed. From the discussion above, I think it's already concluded that this change will *not* be made.